x/cdp/keeper/savings.go

```diff
@@ -0,0 +1,71 @@
+package keeper
+
+import (
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	supplyexported "github.com/cosmos/cosmos-sdk/x/supply/exported"
+	"github.com/kava-labs/kava/x/cdp/types"
+)
+
+// ApplySavingsRate distributes surplus that has accumulated in the liquidator account to address holding stable coins according the the savings rate
+func (k Keeper) ApplySavingsRate(ctx sdk.Context, debtDenom string) sdk.Error {
+	dp, found := k.GetDebtParam(ctx, debtDenom)
+	if !found {
+		return types.ErrDebtNotSupported(k.codespace, debtDenom)
+	}
+	liquidatorMacc := k.supplyKeeper.GetModuleAccount(ctx, types.LiquidatorMacc)
+	totalSurplusCoins := sdk.NewCoins(sdk.NewCoin(debtDenom, liquidatorMacc.GetCoins().AmountOf(dp.Denom)))
+
+	surplusToDistribute := sdk.NewDecFromInt(liquidatorMacc.GetCoins().AmountOf(dp.Denom)).Mul(dp.SavingsRate).RoundInt()
```

Comment:
abci.go passes denom parameter 'usdx' to ApplySavingsRate(debtDenom) but the liquidatorMacc's debt coins are stored under denom 'debt'. Therefore,`liquidatorMacc.GetCoins().AmountOf(dp.Denom)` queries the wrong coin denom, returns 0, and the function returns on line 23 with no savings rate distribution.

Reply:
at least that's my working theory